src/cpu/cpu.cpp

```diff
@@ -2242,39 +2371,115 @@ static void cpu_increase_cycles(bool pressed)
 		LOG_MSG("CPU: max %d percent.", CPU_CyclePercUsed);
 
 	} else {
-		auto old_cycles = CPU_CycleMax;
-		if (cpu_cycle_up < 100) {
-			CPU_CycleMax = static_cast<int>(
-			        CPU_CycleMax *
-			        (1 + static_cast<float>(cpu_cycle_up) / 100.0f));
-		} else {
-			CPU_CycleMax = CPU_CycleMax + cpu_cycle_up;
-		}
-
+		CPU_CycleMax  = calc_cycles_increase(CPU_CycleMax, cpu_cycle_up);
 		CPU_CycleLeft = 0;
 		CPU_Cycles    = 0;
-		if (CPU_CycleMax == old_cycles) {
-			CPU_CycleMax++;
-		}
 
-		if (CPU_CycleMax > 15000) {
-			LOG_MSG("CPU: fixed %d cycles. If you need more than 20000, "
-			        "try core=dynamic in DOSBox's options.",
+		if (CPU_CycleMax > 20000) {
+			LOG_MSG("CPU: Fixed %d cycles If you need more than 20000, "
+			        "try setting 'core = dynamic' for increased performance.",
 			        CPU_CycleMax);
 		} else {
-			LOG_MSG("CPU: fixed %d cycles.", CPU_CycleMax);
+			LOG_MSG("CPU: Fixed %d cycles", CPU_CycleMax);
 		}
```

Comment:
Do we want to also display these messages when using modern settings?

Reply:
Yeah good idea.